ts/Extensions/Breadcrumbs.ts

```diff
@@ -0,0 +1,658 @@
+                    chart.marginBottom) {
+            if (chart.legend && chart.legend.group && chart.legend.options.verticalAlign === 'bottom') {
+                calcPosition.y -= chart.marginBottom -
+                        this.breadcrumbsGroup.getBBox().height -
+                        chart.legend.legendHeight;
+            } else {
+                calcPosition.y -= this.breadcrumbsGroup.getBBox().height;
+            }
+        }
+
+        if (breadcrumbsOptions.position && breadcrumbsOptions.position.align === 'right') {
+            calcPosition.x -= this.breadcrumbsGroup.getBBox().width;
+        } else if (breadcrumbsOptions.position && breadcrumbsOptions.position.align === 'center') {
+            calcPosition.x -= this.breadcrumbsGroup.getBBox().width / 2;
+        }
+        this.breadcrumbsGroup
+            .align(
+                calcPosition,
+                true,
+                chart.spacingBox
+            );
+    }
+
+    /**
+     * This method creates an array of arrays containing a level number
+     * with the corresponding series/point.
+     *
+     * @requires  modules/drilldown
+     *
+     * @function Highcharts.Chart#createList
+     * @param {Highcharts.Breadcrumbs} this
+     *        Breadcrumbs class.
+     */
+    public createList(this: Breadcrumbs): void {
+        const chart = this.chart,
+            drilldownLevels = chart.drilldownLevels,
+            initialSeriesLevel = -1;
+
+        let breadcrumbsList: Array<any> = this.breadcrumbsList || [],
+            // If the list doesn't exist treat the initial series
+            // as the current level- first iteration.
+            currentLevelNumber: number = breadcrumbsList.length ?
+                breadcrumbsList[breadcrumbsList.length - 1][0] : initialSeriesLevel;
+
+        if (drilldownLevels && drilldownLevels.length) {
+            // Add the initial series as the first element.
+            breadcrumbsList = !breadcrumbsList.length ?
+                [[initialSeriesLevel, drilldownLevels[0].seriesOptions]] : breadcrumbsList;
+
+            drilldownLevels.forEach(function (level): void {
+                // If level is already added to breadcrumbs list,
+                // don't add it again- drilling categories
+                if (level.levelNumber > currentLevelNumber) {
+                    currentLevelNumber = level.levelNumber;
+                    breadcrumbsList.push([currentLevelNumber, level.pointOptions]);
+                }
+            });
+        }
+        this.breadcrumbsList = breadcrumbsList;
+    }
+
+    /**
+     * Destroy the chosen buttons and separators.
+     *
+     * @requires  modules/drilldown
+     *
+     * @function Highcharts.Chart#destroy
+     * @param {Highcharts.Breadcrumbs} this
+     *        Breadcrumbs class.
+     * @param {Highcharts.Breadcrumbs} lastVisibleLevel
+     *        Number of levels to destoy.
+     */
+    public destroy(this: Breadcrumbs, lastVisibleLevel: number): void {
```

Comment:
I would also add the possibility when lastVisibleLevel is not set - then it would destroy all the breadcrumbs.